flink-connectors/flink-connector-kafka-base/src/main/java/org/apache/flink/streaming/connectors/kafka/KafkaTableSource.java

```diff
@@ -68,6 +71,12 @@
 	/** Descriptor for a rowtime attribute. */
 	private List<RowtimeAttributeDescriptor> rowtimeAttributeDescriptors;
 
+	/** The startup mode for the contained consumer (default is {@link StartupMode#GROUP_OFFSETS}). */
+	protected StartupMode startupMode;
```

Comment:
should be `private`. We use `protected` setter methods to set optional parameters from the builder (see `proctimeAttribute`).